mitiq/calibration/calibrator.py

```diff
@@ -308,7 +296,7 @@ def get_cost(self) -> Dict[str, int]:
             "ideal_executions": ideal,
         }
 
-    def run(self, log: bool = False) -> None:
+    def run(self, log: bool = False, log_cartesian: bool = False) -> None:
```

Comment:
What if we make `log` a string which can either be `"all"` or `"cartesian"`? Do you think this is a simpler inferface, or more complicated?

Reply:
IMO it wasn't a very good idea to call this parameter "log" because it is not logging but rather the output of calibrations results in some form. 

I have couple ideas regarding this. 

The calibration is nothing more than just running a set of experiments with different executors/circuits/techniques. So in the end you have a bunch of results (now it is 3 numpy arrays) which can be represented in plenty of different ways. We can wrap these arrays into a pandas dataset and have a very convenient sql like query interface for these data. This also can be used as a persistent storage (I am thinking about this https://github.com/unitaryfund/mitiq/issues/1934)

Of course, we can provide a predefined set of the calibration results representations like now it can be just "flat" (not "all") and "cartesian". But still users can use pandas interface to do their own analysis.

I am also thinking about the case when we run all these experiments periodically on various hardware and simulators and see if there are any changes (improvements/degradations). In this case I would like to put the calibration results into a database from where I can easily gather necessary data.

BTW, having old plain logs while calibrator is working can also be useful and it is better to rely here on old plain log level.